### ISSUE TITLE
Setting maxRetries to 0 results in no try at all

### ISSUE THREAD

**Issue by tioback** [elastic/elasticsearch-js#1283]
## 🐛 Bug Report

When working with the Scroll Search Helper, passing the `maxRetries` option as `0` results in no requests being made.

## To Reproduce

```js
const { Client } = require('@elastic/elasticsearch')
const client = new Client({ node: 'http://localhost:9200' })

const scrollSearch = client.helpers.scrollSearch({
  index: 'stackoverflow',
  body: {
    query: {
      match: {
        title: 'javascript'
      }
    }
  }
}, { maxRetries: 0 })

for await (const result of scrollSearch) {
  console.log(result)
}
```
## Expected behavior

I expected the helper to perform at least one attempt, without retrying in case of any error. However, no requests are made.
## Your Environment

- *node version*: 12.14.1
- `@elastic/elasticsearch` *version*: >=7.8.0
- *os*: Mac

**Comment by tioback** [comment]
Upon investigation, I found out that the `for loops` that handle the retries are using a bad compare operator.

I'll open a pull request with a fix proposal.

**Comment by delvedor** [comment]
Closed in #1284.
